### PR TITLE
Added feature for tagging the cache

### PR DESCRIPTION
**Pull request by Nyholm**
I took a lot of inspiration from Laravel.

I've created a new repository with: 
- `TaggableItemInterface`
- `TaggablePoolInterface`
- `TaggablePoolTrait`

They could be used with any other PSR6 cache.

### CODE REVIEW

**Comment by Nyholm** [comment]
Have a look at this readme: https://github.com/php-cache/taggable-cache
The order of the tags matter. That implementation is form Laravel.

Question: Should the order matter?

File: src/CachePool.php [hunk 1]
```diff
@@ -46,16 +51,17 @@ public function __construct(Cache $cache)
     /**
      * {@inheritdoc}
      */
-    public function getItem($key)
+    public function getItem($key, array $tags = array())
```
Comment (cryptiklemur) [inline src/CachePool.php:54]: Doesn't this conflict with the `CacheItemPoolInterface`?

Comment (Nyholm) [inline src/CachePool.php:54]: No, it does not since the second argument is optional. 
If CacheItemPoolInterface were an abstract class it would.

**Comment by Nyholm** [comment]
I cant see why the test are failing on php5.6..
Im running php5.6 on local machine and tests passes.

**Comment by cryptiklemur** [comment]
That is super strange.

**Comment by Nyholm** [comment]
No, changes... now it works.

**Comment by cryptiklemur** [comment]
Yeah, i rebuilt it

**Comment by Nyholm** [comment]
This PR can be merged, Im doing some edge case fixes on https://github.com/php-cache/taggable-cache

**Comment by Nyholm** [comment]
hm.. sorry no,.. still working..

**Comment by Nyholm** [comment]
I found out that I do not need the `TaggableItemInterface`. So I reduced the PR a little. 
Review and merge when you are ready

**Comment by cryptiklemur** [comment]
I'm not sure i like the function tag test

**Comment by aequasi** [comment]
Also, the parent class for that test isn't there

**Comment by Nyholm** [comment]
The functional test is using https://github.com/php-cache/integration-tests (needs to be added to packagist). It makes sure that this repo works with the taggad-cache repo. 

Renaming or refactoring are welcome, but I believe we need this to make sure changes to one repository does not break the other.

**Comment by Nyholm** [comment]
Awesome. Thank you

**Pull request merged**